test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java

```diff
@@ -551,6 +566,232 @@ public void narrativeDecoration() throws Exception {
+        TestCommon.compare(noAuthSearchObjsOut.getObjects().get(0), expected2);
+
+        assertThat("incorrect search objects count", authSearchObjsOut.getObjects().size(), is(2));
+
+        Set<String> res = new HashSet<>();
+        res.add(expected1.getGuid());
+        res.add(expected2.getGuid());
+        assertThat("incorrect results", res.contains(authSearchObjsOut.getObjects().get(0).getGuid()), is(true));
+        assertThat("incorrect results", res.contains(authSearchObjsOut.getObjects().get(1).getGuid()), is(true));
+
+    }
+    @Test
+    public void optionalAuthForSearchTypes() throws Exception {
+        setUpOptionalAuthForSearch();
+
+        SearchTypesOutput authSearchTypesOut;
+        SearchTypesOutput noAuthSearchTypesOut;
+
+        try {
+            noAuthSearchTypesOut = noAuthSearchCli.searchTypes(new SearchTypesInput()
+                    .withMatchFilter(new MatchFilter())
+                    .withAccessFilter(new AccessFilter()));
+
+            authSearchTypesOut = searchCli.searchTypes(new SearchTypesInput()
+                    .withMatchFilter(new MatchFilter())
+                    .withAccessFilter(new AccessFilter()));
+
+        } catch (ServerException e) {
+            System.out.println("Exception server side trace:\n" + e.getData());
+            throw e;
+        }
+
+        Map<String, Long> typeToCount = noAuthSearchTypesOut.getTypeToCount();
+        assertThat("incorrect search types map size", typeToCount.size(), is(1));
+        assertThat("Incorrect search types type name", typeToCount.keySet(), is(set("Deco")));
+        assertThat("Incorrect search types type count", typeToCount.get("Deco"), is(Long.valueOf(1)));
+
+        typeToCount = authSearchTypesOut.getTypeToCount();
+        assertThat("incorrect search types map size", typeToCount.size(), is(1));
+        assertThat("Incorrect search types type name", typeToCount.keySet(), is(set("Deco")));
+        assertThat("Incorrect search types type count", typeToCount.get("Deco"), is(Long.valueOf(2)));
+    }
+
+    @Test
+    public void optionalAuthForGetObjects() throws Exception {
+        setUpOptionalAuthForSearch();
+
+        GetObjectsOutput authGetObjsOut;
+        GetObjectsOutput noAuthGetObjsOut;
+
+        try {
+            noAuthGetObjsOut = noAuthSearchCli.getObjects(new GetObjectsInput()
+                    .withGuids(Arrays.asList("WS:1/1/1", "WS:2/1/1")));
+
+            authGetObjsOut = searchCli.getObjects(new GetObjectsInput()
+                    .withGuids(Arrays.asList("WS:1/1/1", "WS:2/1/1")));
+
+        } catch (ServerException e) {
+            System.out.println("Exception server side trace:\n" + e.getData());
+            throw e;
+        }
+
+        assertThat("incorrect get objects count", noAuthGetObjsOut.getObjects().size(), is(0));
+        assertThat("incorrect get objects count", authGetObjsOut.getObjects().size(), is(2));
+
+    }
+    @Test
```

Comment:
I think this is the old test, right, and so should be deleted?

Reply:
ops